staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -68,6 +72,21 @@ type ObjectTracker interface {
 	Watch(gvr schema.GroupVersionResource, ns string) (watch.Interface, error)
 }
 
+// ManagedFieldObjectTracker keeps track of objects and managed fields. It is intended to be used to
+// fake calls to a server by returning objects based on their kind, namespace and name.
+type ManagedFieldObjectTracker interface {
+	ObjectTracker
+
+	// CreateWithFieldManager adds an object to the tracker in the specified namespace and tracks managed fields.
+	CreateWithFieldManager(gvr schema.GroupVersionResource, obj runtime.Object, ns string, fieldManager string) error
+
+	// UpdateWithFieldManager updates an existing object in the tracker in the specified namespace and tracks managed fields.
+	UpdateWithFieldManager(gvr schema.GroupVersionResource, obj runtime.Object, ns string, fieldManager string) error
+
+	// ApplyWithFieldManager applies an object in the tracker in the specified namespace and tracks managed fields.
+	ApplyWithFieldManager(gvr schema.GroupVersionResource, applyConfiguration runtime.Object, ns string, fieldManager string, force bool) error
```

Comment:
Rather than FieldManager-specific interface, I think there may be an argument for plumping the `PatchOptions` and `UpdateOptions`, etc. up through the `ObjectTracker` signatures for any other options we may want to add support for in the future.

Reply:
I do like the idea of augmenting `ObjectTracker` with options.  Does it sound reasonable to make breaking changes to the `ObjectTracker` interface.  It's possible that someone might have a custom implementation, but is seems like a rather bespoke thing to do.

Reply:
I know there are some direct calls to the interface methods within tests. I think it would be better to include the options in the interface itself. If we are concerned about third party breaking changes it can be made into an optional/variadic argument.

Reply:
SGTM. I'm going to use the varadic argument since there is this annoying directly call to `ObjectTracker.Watch` in the generated clients, so if I change that signature, I break skew support between generated code (generated using kube 1.30 or earlier) and the fixture code.